doc/api/child_process.markdown

```diff
@@ -1,359 +1,107 @@
-a `'message'` event on the child.
-
-For example:
-
-    const cp = require('child_process');
-    const n = cp.fork(`${__dirname}/sub.js`);
-
-    n.on('message', (m) => {
-      console.log('PARENT got message:', m);
+    ls.stderr.on('data', (data) => {
+      console.log(`stderr: ${data}`);
     });
 
-    n.send({ hello: 'world' });
-
-And then the child script, `'sub.js'` might look like this:
-
-    process.on('message', (m) => {
-      console.log('CHILD got message:', m);
+    ls.on('close', (code) => {
+      console.log(`child process exited with code ${code}`);
     });
 
-    process.send({ foo: 'bar' });
-
-In the child, the `process` object will have a `send()` method, and `process`
-will emit objects each time it receives a message on its channel.
-
-There is a special case when sending a `{cmd: 'NODE_foo'}` message. All messages
-containing a `NODE_` prefix in its `cmd` property will not be emitted in
-the `'message'` event, since they are internal messages used by Node.js core.
-Messages containing the prefix are emitted in the `'internalMessage'` event.
-Avoid using this feature; it is subject to change without notice.
-
-The `sendHandle` option to `child.send()` is for sending a TCP server or
-socket object to another process. The child will receive the object as its
-second argument to the `'message'` event.
+By default, pipes for `stdin`, `stdout` and `stderr` are established between
+the parent Node.js process and the spawned child. It is possible to stream data
+through these pipes in a non-blocking way. *Note, however, that some programs
+use line-buffered I/O internally. While that does not affect Node.js, it can
+mean that data sent to the child process may not be immediately consumed.*
+
+For convenience, the `child_process` module provides a handful of synchronous
+and asynchronous alternatives to `child_process.spawn()`, each of which are
+documented fully [below][]:
+
+  * `child_process.exec()`: spawns a shell and runs a command within that shell,
+    passing the `stdout` and `stderr` to a callback function when complete.
+  * `child_process.execFile()`: similar to `child_process.exec()` except that
+    it spawns the command directly without first spawning a shell.
+  * `child_process.fork()`: spawns a new Node.js process and invokes a
+    specified module with an IPC communication channel established that allows
+    sending messages between parent and child.
+  * `child_process.spawnSync()`: a synchronous version of   
+    `child_process.spawn()` that *will* block the Node.js event loop.
+  * `child_process.execSync()`: a synchronous version of
+    `child_process.exec()` that *will* block the Node.js event loop.
+  * `child_process.execFileSync()`: a synchronous version of
+    `child_process.execFile()` that *will* block the Node.js event loop.
+
+For certain use cases, such as automating shell scripts, the
+[synchronous counterparts][] may be more convenient.
```

Comment:
For all other cases, particular production servers, the Sync counterparts stall the event loop, causing poor performance our outright bugs. Consider carefully whether they are appropriate before using.

^--- maybe?